api/metrics/views.py

```diff
@@ -154,3 +160,42 @@ def metric_type(self):
     @property
     def metric(self):
         return PreprintDownload
+
+class RawMetricsView(GenericAPIView):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        IsRawMetricsUser,
+        TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.METRICS_BASIC]
+    required_write_scopes = [CoreScopes.METRICS_RESTRICTED]
+
+    view_category = 'raw-metrics'
+    view_name = 'raw-metrics-view'
+
+    serializer_class = RawMetricsSerializer
+
+    @require_switch(ENABLE_RAW_METRICS)
+    def delete(self, request, *args, **kwargs):
+        raise ValidationError('DELETE not supported. Use GET/POST/PUT')
+
+    @require_switch(ENABLE_RAW_METRICS)
+    def get(self, request, *args, **kwargs):
```

Comment:
Can we just redirect all of these using django? I'm not sure if that will work with the post methods though.

Reply:
My understanding was that this needed to be a proxy endpoint and not a redirect since the elasticsearch endpoint isn't exposed.

Reply:
yeah it shouldn't really make a difference. I just thought you'd be able to redirect the request and just put additional permissions on it, but this is functionally the same thing.